lib/beaker-hostgenerator/parser.rb

```diff
@@ -225,6 +235,11 @@ def settings_string_to_map(host_settings)
           raise BeakerHostGenerator::Exceptions::InvalidNodeSpecError,
                 "Malformed host settings: #{host_settings}"
         end
+        if pair[1].include?("[")
+          pair[1] = pair[1].
+          delete('[]').
+          split('~')
+        end
```

Comment:
I don't think this needs extra validation as anything within the `[]` should technically work.